develop-docs/backend/api/design.mdx

```diff
@@ -1,57 +1,154 @@
-for building Web APIs. Sentry APIs are built using DRF. Here are some considerations to make when designing APIs at Sentry.
+This document contains a set of design principles and requirements whish should be applied to all Sentry APIs. These are _requirements_ when designing public APIs, but internal APIs should attempt to adhere to these principles as well.
 
-## URL Patterns
-The API's URL is what developers use to call the endpoint so it’s important that it is meaningful and clear.
+In the Sentry monolith, we use [Django REST framework](https://www.django-rest-framework.org/) (DRF).
 
-### Don't Exceed 3-level Nesting
-Nested resources format like `api/0/organizations/\{org}/projects/` is recommended over `api/0/projects/` for readability because it gives user an understanding of resource hierarchy. However, nesting can make URLs too long and hard to use. Sentry uses 3-level nesting as a hybrid solution.
+<Alert>
 
-Here are some possible urls for values with this resource hierarchy: organization -> project -> tag -> value
-- 👍 `/projects/\{organization_slug}/\{project_slug}/tags/\{tag_id}/values`
-- 👎 `/organizations/\{organization_slug}/projects/\{project_slug}/tags/\{tag_id}/values/`
-- 👎 `/values/`
+**Note:** Not every API we have exposed follows these guidelines, but is expected that new endpoints do, and we will work to migrate existing endpoints to adhere to these principles.
+
+</Alert>
+
+## APIs are Public
+
+**Every endpoint in Sentry's API is public.**
+
+You should design it as such. You should **not** expose internal designs, including parameters which control behaviors of the underlying system.
+
+For example, if you had an endpoint that needs to query a replica in some cases, do **not** expose a parameter to control the behavior (e.g. `useReplica=1`). Instead you should control this flag on the backend based on other information (such as a feature flag or account setting).
+
+## Route Design
+
+Routes are one of the most important parts of your API structure, as they communicate relationships and intent. Its important to get them right and adhere to a set of guidelines to make it easier to understand and use the API.
+
+This is no different - and just as important - to how you think about the URL structure of your user interface.
+
+Use the following guidelines for naming resources and their collections:
+
+- **Do** use lowercase and hyphenated collection names, e.g. `commit-files`.
+- **Do** use plural collection names. Avoid using uncountable words because the user can't know whether the GET returns one item or a list.
+- **Do** use `snake_case` for route parameters. e.g. `tags/\{tag_name}/`.
+- **Do** consistently shorten parameters that are excessively long when the term will unambiguous. e.g. `organization` -> `org`.
 
-In the above example we flattened `projects`. The table below shows the existing flattened collections which works out with our existing APIs.
+Standard parameters that should be shortened in route names:
```

Comment:
Is it worth clarifying what route name means here? Is it the path segments, parameters in the route, or the internal name that is used for route resolution?